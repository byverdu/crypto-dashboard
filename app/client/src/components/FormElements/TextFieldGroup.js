import React from 'react';
import * as UI from '@material-ui/core/';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

const styles = {};

const TextFieldGroup = ({
  handleChangeInput, formData
}) => (
  <UI.FormControl component="fieldset" required>
    <UI.FormLabel component="legend">Select Asset and Amount</UI.FormLabel>

    {formData && formData.map( prop => (
      <UI.TextField
        key={prop.name}
        {...prop}
        onChange={handleChangeInput}
      />
    ))}
  </UI.FormControl>
);

TextFieldGroup.propTypes = {
  handleChangeInput: PropTypes.func.isRequired,
  formData: PropTypes.array.isRequired,
  classes: PropTypes.any
};

export default withStyles( styles )( TextFieldGroup );
