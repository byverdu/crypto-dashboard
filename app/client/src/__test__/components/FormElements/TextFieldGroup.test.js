import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TextFieldGroup from '../../../components/FormElements/TextFieldGroup';

const formData = require( '../../../config/data' ).inputFields;

let wrapped;
const component = <TextFieldGroup handleChangeInput={() => {}} formData={formData} />;

beforeEach(() => {
  wrapped = shallow( component );
});

describe( 'RadioGroup component', () => {
  it( 'renders correctly', () => {
    const tree = renderer
      .create( component )
      .toJSON();

    expect( tree ).toMatchSnapshot();
  });

  it( 'should have a "handleChangeInput" prop', () => {
    expect( wrapped.props()).toHaveProperty( 'handleChangeInput' );
    expect( typeof wrapped.props().handleChangeInput ).toEqual( 'function' );
  });

  it( 'should have a "formData" prop', () => {
    expect( wrapped.props()).toHaveProperty( 'formData' );
    expect( wrapped.props().formData.length ).toBeGreaterThanOrEqual( 2 );
  });
});
