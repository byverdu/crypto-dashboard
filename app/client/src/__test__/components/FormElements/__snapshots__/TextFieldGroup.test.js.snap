// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup component renders correctly 1`] = `
<fieldset
  className="MuiFormControl-root-1"
>
  <legend
    className="MuiFormLabel-root-5 MuiFormLabel-required-10"
  >
    Select Asset and Amount
    <span
      className="MuiFormLabel-asterisk-11"
    >
       *
    </span>
  </legend>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-1"
    step="0.0001"
  >
    <label
      className="MuiFormLabel-root-5 MuiFormLabel-required-10 MuiInputLabel-root-12 MuiInputLabel-formControl-13 MuiInputLabel-animated-16 MuiInputLabel-outlined-18"
      data-shrink={false}
      htmlFor="amountCrypto"
    >
      Amount Crypto
      <span
        className="MuiFormLabel-asterisk-11"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-32 MuiOutlinedInput-root-19 MuiInputBase-formControl-33"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="MuiPrivateNotchedOutline-root-49 MuiOutlinedInput-notchedOutline-26"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="MuiPrivateNotchedOutline-legend-50"
          style={
            Object {
              "width": 0.01,
            }
          }
        />
      </fieldset>
      <input
        aria-invalid={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInputBase-input-42 MuiOutlinedInput-input-27 MuiInputBase-inputType-45"
        defaultValue={undefined}
        disabled={false}
        id="amountCrypto"
        name="amountCrypto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Add coin amount"
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="number"
        value={undefined}
      />
    </div>
  </div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-1"
    step="0.0001"
  >
    <label
      className="MuiFormLabel-root-5 MuiInputLabel-root-12 MuiInputLabel-formControl-13 MuiInputLabel-animated-16 MuiInputLabel-outlined-18"
      data-shrink={false}
      htmlFor="priceCrypto"
    >
      Price Crypto
    </label>
    <div
      className="MuiInputBase-root-32 MuiOutlinedInput-root-19 MuiInputBase-formControl-33"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="MuiPrivateNotchedOutline-root-49 MuiOutlinedInput-notchedOutline-26"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="MuiPrivateNotchedOutline-legend-50"
          style={
            Object {
              "width": 0.01,
            }
          }
        />
      </fieldset>
      <input
        aria-invalid={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInputBase-input-42 MuiOutlinedInput-input-27 MuiInputBase-inputType-45"
        defaultValue={undefined}
        disabled={false}
        id="priceCrypto"
        name="priceCrypto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Add price coin"
        readOnly={undefined}
        required={false}
        rows={undefined}
        type="number"
        value={undefined}
      />
    </div>
  </div>
</fieldset>
`;
